StreamChatClient_v3/Workers/Worker.swift

```diff
@@ -0,0 +1,18 @@
+//
+// Worker.swift
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+typealias WorkerBuilder = (
+  _ storageContext: NSManagedObjectContext,
+  _ webSocketClient: WebSocketClient,
+  _ apiClient: APIClient
+) -> Worker
```

Comment:
The `storageContext` should be just `Storage` to avoid the distribution of storage functionality through other files.

Reply:
Every working has its own context to work on to make sure the changes are transactional and that they can each work on a separate thread. There's definitely room for improvement but I'd worry about that once we have more code in.

Reply:
Can't we wrap this to encapsulate CoreData?

Reply:
I think you can wrap it as @b-onc suggested.

Reply:
Probably you already did that in other branches.